sound/soc/sof/intel/hda-dai.c

```diff
@@ -289,7 +289,6 @@ static int hda_dai_trigger(struct snd_pcm_substream *substream, int cmd, struct
 
 	switch (cmd) {
 	case SNDRV_PCM_TRIGGER_SUSPEND:
-	case SNDRV_PCM_TRIGGER_STOP:
 		ret = hda_link_dma_cleanup(substream, hext_stream, dai);
```

Comment:
hda_link_dma_prepare is called by ASoC framework, not by widget setup. If we don't clean up hda_link_dma, do we need to skip some hda_link_dma_prepare ?

Reply:
not following @RanderWang 's question. Can you please elaborate?

Reply:
what do you mean by hda_link_dma_prepare? There is no such function? Anyway, if I can guess what you're asking, the stream tag is assigned in hda_link_hw_params() and with a repeated start, if the stream tag is not freed, prepare wouldnt really do anything, thus preserving the original stream tag for the start trigger

Reply:
o, my code is out of date.  I can't find it after update code